### PR TITLE
[tests] Enable MLIR C++ unittest with lit

### PR DESCRIPTION
**Pull request by Menooker**
Add an example gtest unittest for MLIR and let it be driven by lit. The required cmake and lit configs are also added.

### CODE REVIEW

**Comment by Menooker** [comment]
This PR requires a new build of LLVM `-DLLVM_INSTALL_GTEST=ON`, otherwise it will fail to find gtest header and static library. How do I let CI re-build LLVM?

**Review by ZhennanQin — APPROVED** [review]

File: cmake/llvm-version.txt [hunk 1]
```diff
@@ -1 +1 @@
-179e174945b6c0da462c534504720c9544aebf84
+5adcfd4c17826b2b8f023881baa1c7f79cb23920
```
Comment (ciyongch) [inline cmake/llvm-version.txt:1]: It should be fine to combine the llvm version update and other changes in a single PR at this stage, but it'd be better to split this in future?

File: test/gc-dialects/Unit/lit.cfg.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# -*- Python -*-
```
Comment (ciyongch) [inline test/gc-dialects/Unit/lit.cfg.py:1]: Maybe we don't need the multiple directories like `gc-dialects` to share the `lit` python files for other components like transforms/passes, runtime and others?

Comment (Menooker) [reply test/gc-dialects/Unit/lit.cfg.py:1]: The C++-based unittest is a special case. Filecheck tests don't need this lit config file.

File: cmake/llvm-version.txt (hunk 1 above)
Comment (Menooker) [reply cmake/llvm-version.txt:1]: It is to trigger LLVM rebuilt in CI. We need a new flag in cmake options of LLVM.

**Review by ciyongch — APPROVED** [review]

**Pull request merged**